plutus-core/plutus-core/src/PlutusCore/DeBruijn/Internal.hs

```diff
@@ -53,48 +56,64 @@ import Data.Word
 import Prettyprinter
 
 import Control.DeepSeq (NFData)
+import Data.Coerce
 import ErrorCode
 import GHC.Generics
 
 -- | A relative index used for de Bruijn identifiers.
 newtype Index = Index Word64
-    deriving stock Generic
-    deriving newtype (Show, Num, Enum, Real, Integral, Eq, Ord, Pretty)
-    deriving anyclass NFData
+    deriving stock (Generic)
+    deriving newtype (Show, Num, Enum, Real, Integral, Eq, Ord, Pretty, NFData)
 
 -- | The LamAbs index (for debruijn indices) and the starting level of DeBruijn monad
 deBruijnInitIndex :: Index
 deBruijnInitIndex = 0
 
 -- | A term name as a de Bruijn index.
 data NamedDeBruijn = NamedDeBruijn { ndbnString :: T.Text, ndbnIndex :: Index }
-    deriving (Show, Generic)
+    deriving stock (Show, Generic)
     deriving anyclass NFData
 
+{-| A newtype that has a `fakeName` (missing name).
+It is used for:
+1) testing purposes
+2) speeding up the script CEK-execution (see `Plutus.V1.Ledger.Api.ScriptForExecution`)
```

Comment:
This isn't quite right: it doesn't speed up _execution_. The thing that's special about it is we can give a specialized serialization instance that behaves the same as a nameless de Bruijn name.